dht_test.go

```diff
@@ -1910,3 +1910,51 @@ func TestProtocolUpgrade(t *testing.T) {
 		t.Fatalf("Expected 'buzz' got '%s'", string(value))
 	}
 }
+
+func TestInvalidKeys(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	nDHTs := 2
+	dhts := setupDHTS(t, ctx, nDHTs)
+	defer func() {
+		for i := 0; i < nDHTs; i++ {
+			dhts[i].Close()
+			defer dhts[i].host.Close()
```

Comment:
the need for this defer seems like a bit of black magic, no? is there a reason the hosts need to all be closed after the dhts?

Reply:
Maybe? I really don't know. I'm just copying other tests.